core/tests/integration/inputs/typecheck/chaining_dictionary_function_record_dictionary_subtyping.ncl

```diff
@@ -0,0 +1,8 @@
+# test.type = 'error'
+# [test.metadata]
+# error = 'EvalError::BlameError'
+let x = { foo = 1, bar =2} : {foo : Number, bar : Number} in
+let y = (std.record.insert "baz" 5 x) : { _ : Number } in
+let _ = (std.record.get "foo" y) : Number in
+let z = (std.record.remove "foo" y) : { _ : Number } in
+std.record.get "foo" z 
```

Comment:
I believe all the type annotations are exactly the type that should be inferred, so they can probably be omitted.

```suggestion
let result : _ =
  {foo = 1, bar = 2}
  |> std.record.insert "baz" 5 x
  |> std.record.get "foo" y
  |> std.record.remove "foo" y
  |> std.record.get "foo" z
in
result
```